go-app.js

```diff
@@ -25,100 +24,115 @@ go.app = function() {
+            })
+            .then(function(resp) {
+                return self.states.create('states:exit', { echo: resp.data});
+            });
     });
 
-    self.states.add('states:error', function(name) {
-        // When we get a response with a status code that is not in the 200
-        // range, an error is thrown. We use `.catch()` (the promise method to
-        // use for error callbacks) to handle the error instead of `.then()`
-        // (the promise method to use for success callbacks).  The error object
-        // given as a parameter to `.catch()` is an HttpResponseError instance,
-        // so we can access the response off the object as `.response`. Note
-        // that if we wanted to have both success and error callbacks, we could
-        // provide both as arguments to `.then()` instead of using `.catch()`.
+    self.states.add('states:purple', function(name) {
         return self
-            .http.get('http://httpbin.org/status/418')
-            .catch(function(e) {
-                return new EndState(name, {
-                    text: [
-                        "You just performed a request. It got a response",
-                        "with the status code", e.response.code
-                    ].join(' '),
-                    next: 'states:start'
-                });
+            .http.get(
+                'http://lapi.transitchicago.com/api/1.0/ttpositions.aspx?key=33305d8dcece4aa58c651c740f88d1e2&rt=p&outputType=JSON', {
+                params: {rt: 'p', key: '33305d8dcece4aa58c651c740f88d1e2', outputType: 'JSON'}
+            })
+            .then(function(resp) {
+                return self.states.create('states:exit', { echo: resp.data});
             });
     });
+
+    self.states.add('states:pink', function(name) {
+        return self
+            .http.get(
+                'http://lapi.transitchicago.com/api/1.0/ttpositions.aspx?key=33305d8dcece4aa58c651c740f88d1e2&rt=pink&outputType=JSON', {
+                params: {rt: 'pink', key: '33305d8dcece4aa58c651c740f88d1e2', outputType: 'JSON'}
+            })
+            .then(function(resp) {
+                return self.states.create('states:exit', { echo: resp.data});
+            });
+    });
+
+    // self.states.add('states:yellow', function(name) {
+    //     return self
+    //         .http.get(
+    //             'http://lapi.transitchicago.com/api/1.0/ttpositions.aspx?key=33305d8dcece4aa58c651c740f88d1e2&rt=y&outputType=JSON', {
+    //             params: {rt: 'y', key: '33305d8dcece4aa58c651c740f88d1e2', outputType: 'JSON'}
+    //         })
+    //         .then(function(resp) {
+    //             return self.states.create('states:exit', { echo: resp.data});
+    //         });
+    // });
+
+    self.states.add('states:exit', function(name) {
+        return new EndState(name, {
+            text: 'Thanks for using CTA tran tracker.',
+            next: 'states:start'
+        });
+    });
```

Comment:
This means that we don't use the results of the HTTP request. We should probably inform the user of the results in some way. Is that planned for a future PR?